rero_ils/modules/items/serializers/collector.py

```diff
@@ -297,81 +297,92 @@ def _append_res_local_fields(resource_type, resource_pid, csv_data):
-                item_pid=A(
-                    'terms', field='loan.item.pid',
-                    aggs=dict(last_op=A('max', field='date')))))
-        query.aggs.bucket('checkout', checkout_agg)
-        # adds renewal aggregation
-        renewal_agg = A(
-            'filter', term={'loan.trigger': ItemCirculationAction.EXTEND},
-            aggs=dict(item_pid=A('terms', field='loan.item.pid')))
-        query.aggs.bucket('renewal', renewal_agg)
-        # adds last transaction aggregation for the fours triggers below.
-        triggers = [
-            ItemCirculationAction.CHECKOUT,
-            ItemCirculationAction.CHECKIN,
-            ItemCirculationAction.EXTEND
-        ]
-        loans_agg = A('filter', terms={'loan.trigger': triggers},
-                      aggs=dict(
-            item_pid=A('terms',
-                       field='loan.item.pid',
-                       aggs=dict(last_op=A('max', field='date')))))
-        query.aggs.bucket('loans', loans_agg)
-        # query execution
-        result = query.execute()
-        # dump output into a dict
-        # checkouts data
-        loans = {
-            term.key: {'checkout_count': term.doc_count,
-                       'last_checkout': ciso8601.parse_datetime(
+        def _get_loans_by_item_pids(pids):
+            # initial es query to return all loans for the given item_pids
+            query = OperationLogsSearch()\
+                .filter('terms', loan__item__pid=pids)
+            # adds checkouts aggregation
+            checkout_agg = A(
+                'filter',
+                term={'loan.trigger': ItemCirculationAction.CHECKOUT},
+                aggs=dict(
+                    item_pid=A(
+                        'terms', field='loan.item.pid',
+                        size=chunk_size,
+                        aggs=dict(last_op=A('max', field='date')))))
+            query.aggs.bucket('checkout', checkout_agg)
+            # adds renewal aggregation
+            renewal_agg = A(
+                'filter',
+                term={'loan.trigger': ItemCirculationAction.EXTEND},
+                aggs=dict(item_pid=A(
+                    'terms', size=chunk_size, field='loan.item.pid')))
+            query.aggs.bucket('renewal', renewal_agg)
+            # adds last transaction aggregation for the fours triggers below.
+            triggers = [
+                ItemCirculationAction.CHECKOUT,
+                ItemCirculationAction.CHECKIN,
+                ItemCirculationAction.EXTEND
+            ]
+            loans_agg = A(
+                'filter', terms={'loan.trigger': triggers}, aggs=dict(
+                    item_pid=A(
+                        'terms',
+                        field='loan.item.pid',
+                        size=chunk_size,
+                        aggs=dict(last_op=A('max', field='date')))))
+            query.aggs.bucket('loans', loans_agg)
+            # query execution
+            result = query.execute()
```

Comment:
Only aggregations seems used. you could set size of the results to 0
`result = query[0:0].execute()`